lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/Pure/State/Payment.hs

```diff
@@ -55,15 +78,108 @@ import qualified Data.Text as T
 
 data ErrCreatePayment
     = ErrCreatePaymentNotRecentEra (Read.EraValue Read.Era)
-    | ErrCreatePaymentBalanceTx (Write.ErrBalanceTx Write.Conway)
+    | ErrNotEnoughAda { shortfall :: Value }
```

Comment:
Only providing the `shortfall` field allows us to merge `Write.ErrBalanceTxAssetsInsufficient` and `Write.ErrBalanceTxUnableToCreateChange`, which _seems_ desirable. We'd probably ideally want some more info though.